### PR TITLE
snmp_server: Resource modules

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY

Snmp server Resource modules

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by ashwini-mhatre — COMMENTED** [review]
Add new module entry in runtime.yaml

File: plugins/module_utils/network/ios/rm_templates/snmp_server.py [hunk 1]
```diff
@@ -0,0 +1,1720 @@
...
+            cmd += "snmp-server enable traps bgp"
+        if conf.get("cbgp2"):
+            cmd += " cbgp2"
+        if conf.get("state_changes"):
+            if conf.get("state_changes").get("enable"):
+                cmd += " state-changes"
+            if conf.get("state_changes").get("all"):
+                cmd += " all"
+            if conf.get("state_changes").get("backward_trans"):
+                cmd += " backward-trans"
```
Comment (ashwini-mhatre) [inline plugins/module_utils/network/ios/rm_templates/snmp_server.py:100]: is state-changes missing in cmd formation?

File: plugins/module_utils/network/ios/rm_templates/snmp_server.py [hunk 2]
```diff
@@ -0,0 +1,1720 @@
...
+        {
+            "name": "traps.energywise",
+            "getval": re.compile(
+                r"""
+                ^snmp-server\senable\straps\senergywise$
+                """, re.VERBOSE),
+            "setval": "snmp-server enable traps energywise",
+            "result": {
+                "traps": {
+                    "energywise": True,
```
Comment (ashwini-mhatre) [inline plugins/module_utils/network/ios/rm_templates/snmp_server.py:805]: is this always True?

Comment (KB-perByte) [reply plugins/module_utils/network/ios/rm_templates/snmp_server.py:805]: yes, if the parser hits a match it is supposed to be true.

File: plugins/module_utils/network/ios/rm_templates/snmp_server.py (hunk 1 above)
Comment (KB-perByte) [reply plugins/module_utils/network/ios/rm_templates/snmp_server.py:100]: it is added as per the expected command formation in ln 95

**Review by ashwini-mhatre — APPROVED** [review]

**Comment by KB-perByte** [comment]
recheck

**Comment by KB-perByte** [comment]
recheck

**Comment by KB-perByte** [comment]
recheck

**Comment by KB-perByte** [comment]
recheck

**Pull request merged**